src/sources/soundsourceffmpeg.cpp

```diff
@@ -1205,93 +1179,190 @@ ReadableSampleFrames SoundSourceFFmpeg::readSampleFramesClamped(
+                            getSignalInfo().frames2samples(missingFrameRange.length());
                     if (pOutputSampleBuffer) {
                         SampleUtil::clear(
                                 pOutputSampleBuffer,
                                 clearSampleCount);
                         pOutputSampleBuffer += clearSampleCount;
                     }
-                    writableFrameRange.shrinkFront(clearFrameRange.length());
+                    writableFrameRange.shrinkFront(clearFrameCount);
+                }
+            }
+            DEBUG_ASSERT(writableFrameRange.empty() ||
+                    writableFrameRange.start() >= readFrameIndex);
+
+            // -= 2nd step =-
+            // Check for skipped sample data and log
+            DEBUG_ASSERT(readFrameIndex <= decodedFrameRange.start());
+            const auto skippedFrameRange =
+                    IndexRange::between(
+                            readFrameIndex,
+                            decodedFrameRange.start());
+            if (!skippedFrameRange.empty()) {
+                // The decoder has skipped some sample data that needs to
+                // be filled with silence to continue decoding! This is supposed
+                // to occur only at the beginning of a stream for the very first
+                // decoded frame with a lead-in due to start_time > 0. But not all
+                // encoded streams seem to account for this by correctly setting
+                // the start_time property.
+                // NOTE: Decoding might even start at a negative position for the
+                // first frame of the file, i.e. outside of the track's valid range!
+                // Consequently isValidFrameIndex(readFrameIndex) might return false.
+                // This is expected behavior and will be compensated during 'preskip'
+                // (see below).
+                if (readFrameIndex <= frameIndexRange().start()) {
+                    kLogger.debug()
+                            << "Generating silence for skipped sample data"
+                            << skippedFrameRange
+                            << "at the start of the audio stream";
+                } else {
+                    kLogger.warning()
+                            << "Generating silence for skipped sample data"
+                            << skippedFrameRange;
                 }
             }
 
-            // Skip all missing and decoded ranges that do not overlap
+#if VERBOSE_DEBUG_LOG
+            kLogger.debug()
+                    << "Before discarding excessive sample data:"
+                    << "m_curFrameIndex" << m_curFrameIndex
+                    << "readFrameIndex" << readFrameIndex
+                    << "decodedFrameRange" << decodedFrameRange
+                    << "writableFrameRange" << writableFrameRange;
+#endif
+
+            // -= 3rd step =-
+            // Discard both skipped and decoded frames that do not overlap
```

Comment:
What does overlap mean?

Reply:
as the comment states: "...with writableFrameRange, i.e. that precede writableFrameRange"

writeFrameRange is the target, everything else is the source.